fdbclient/MonitorLeader.actor.cpp

```diff
@@ -169,7 +169,7 @@ void ClusterConnectionString::resolveHostnamesBlocking() {
 }
 
 void ClusterConnectionString::resetToUnresolved() {
-	if (hostnames.size() > 0) {
+	if (status == RESOLVED && hostnames.size() > 0) {
```

Comment:
This has no effect when status == RESOLVING. Is this expected?

Reply:
This is intended, and is to fix a corner case. Imagine thread 1 enters the resolve actor, and wait for the resolving to finish, so there could be a context switch. Then thread 2 calls this resetToUnresolved(), which will reset status to UNRESOLVED immediately. Before thread 1 finishes, thread 3 enters the resolve actor, so 1 and 3 will add duplicated result. I found this bug during manual test.